packages/react-instantsearch-dom/src/hooks/useAnswers.js

```diff
@@ -0,0 +1,99 @@
+import React, { useState, useEffect, useMemo, useContext } from 'react';
+import { instantSearchContext } from 'react-instantsearch-core';
+import { createConcurrentSafePromise } from '../lib/createConcurrentSafePromise';
+import { debounce } from '../lib/debounce';
+
+function hasReactHooks() {
+  // >= 16.8.0
+  const [major, minor] = React.version.split('.').map(Number);
+  return major >= 17 || (major === 16 && minor >= 8);
+}
+
+export default function useAnswers({
+  searchClient,
+  queryLanguages,
+  attributesForPrediction,
+  nbHits,
+  renderDebounceTime = 100,
+  searchDebounceTime = 100,
+  ...extraParameters
+}) {
+  if (!hasReactHooks()) {
+    throw new Error(
+      `\`Answers\` component and \`useAnswers\` hook require all React packages to be 16.8.0 or higher.`
+    );
+  }
+  const context = useContext(instantSearchContext);
+  const [query, setQuery] = useState();
+  const [index, setIndex] = useState();
```

Comment:
Why don't we set the initial value to `context.mainTargetedIndex` here?

Reply:
It makes sense https://github.com/algolia/react-instantsearch/pull/2996/commits/0228c51c2d65e477fbce6471e04a85597f5bfda6

Reply:
shouldn't that take IndexContext in account too?

Reply:
Yes, indeed!

Reply:
IIUC, it's related to https://github.com/algolia/react-instantsearch/pull/2996#discussion_r565490874

In the next PR, I will expose `indexContext` as I exposed `instantSearchContext` and I will take the indexName from either of contexts. Does it make sense to you?

Reply:
Yes, as you prefer!